cmd/kubeadm/app/cmd/upgrade/plan.go

```diff
@@ -52,21 +59,78 @@ func NewCmdPlan(apf *applyPlanFlags) *cobra.Command {
 				return err
 			}
 
-			return runPlan(flags, userVersion)
+			printer, err := outputFlags.ToPrinter()
+			if err != nil {
+				return err
+			}
+
+			return runPlan(flags, userVersion, printer)
 		},
 	}
 
+	outputFlags.AddFlags(cmd)
+
 	// Register the common flags for apply and plan
 	addApplyPlanFlags(cmd.Flags(), flags.applyPlanFlags)
 	return cmd
 }
 
+// newUpgradePlanItem helper creates outputapiv1alpha1.UpgradePlanItem object
+func newUpgradePlanItem(component, currentVersion, newVersion string) *outputapiv1alpha1.UpgradePlanItem {
+	return &outputapiv1alpha1.UpgradePlanItem{
+		Component:      component,
+		CurrentVersion: currentVersion,
+		NewVersion:     newVersion,
+	}
+}
+
+// upgradePlanTextPrinter prints upgrade plan in a text form
+type upgradePlanTextPrinter struct {
+	output.TextPrinter
+	columns   []string
+	tabwriter *tabwriter.Writer
+}
+
+// Flush writes any buffered data
+func (uptp *upgradePlanTextPrinter) Flush() {
+	if uptp.tabwriter != nil {
+		uptp.tabwriter.Flush()
+		uptp.tabwriter = nil
+	}
+}
+
+// PrintObj is an implementation of ResourcePrinter.PrintObj for upgrade plan plain text output
+func (uptp *upgradePlanTextPrinter) PrintObj(obj runtime.Object, writer io.Writer) error {
+	if uptp.tabwriter == nil {
+		uptp.tabwriter = tabwriter.NewWriter(writer, 10, 4, 3, ' ', 0)
+		// Print header
+		fmt.Fprintln(uptp.tabwriter, strings.Join(uptp.columns, "\t"))
+	}
+
+	// Print item
+	item := obj.(*outputapiv1alpha1.UpgradePlanItem)
```

Comment:
It's best to check if the cast was good. Something spurious may surprise us here and panic will ensue.

Reply:
makes sense, thank you. will do.